subsys/bluetooth/audio/csis_client.c

```diff
@@ -140,6 +143,110 @@ static struct bt_csis *lookup_instance_by_set_info(const struct bt_csis_client_s
 	return NULL;
 }
 
+static int member_rank_compare_asc(const void *m1, const void *m2)
+{
+	struct bt_csis_client_set_member *member_1 = *(struct bt_csis_client_set_member **)m1;
+	struct bt_csis_client_set_member *member_2 = *(struct bt_csis_client_set_member **)m2;
+	struct bt_csis *inst_1;
+	struct bt_csis *inst_2;
+
+	/* TODO: lookup_instance_by_set_info should be able to work on a const
+	 * member, but it relies on `bt_conn_index` which doesn't take a const
+	 * bt_conn.
+	 */
+
+	inst_1 = lookup_instance_by_set_info(member_1, active.info);
+	inst_2 = lookup_instance_by_set_info(member_2, active.info);
+
+	if (inst_1 == NULL) {
+		BT_ERR("inst_1 was NULL for member %p", member_1);
+		return 0;
+	}
+
+	if (inst_2 == NULL) {
+		BT_ERR("inst_2 was NULL for member %p", member_2);
+		return 0;
+	}
+
+	if (inst_1->cli.rank < inst_2->cli.rank) {
+		return -1;
+	} else if (inst_1->cli.rank > inst_2->cli.rank) {
+		return 1;
+	}
+
+	return 0;
+}
+
+static int member_rank_compare_desc(const void *m1, const void *m2)
+{
+	struct bt_csis_client_set_member *member_1 = *(struct bt_csis_client_set_member **)m1;
+	struct bt_csis_client_set_member *member_2 = *(struct bt_csis_client_set_member **)m2;
+	struct bt_csis *inst_1;
+	struct bt_csis *inst_2;
+
+	/* TODO: lookup_instance_by_set_info should be able to work on a const
+	 * member, but it relies on `bt_conn_index` which doesn't take a const
+	 * bt_conn.
+	 */
+
+	inst_1 = lookup_instance_by_set_info(member_1, active.info);
+	inst_2 = lookup_instance_by_set_info(member_2, active.info);
+
+	if (inst_1 == NULL) {
+		BT_ERR("inst_1 was NULL for member %p", member_1);
+		return 0;
+	}
+
+	if (inst_2 == NULL) {
+		BT_ERR("inst_2 was NULL for member %p", member_2);
+		return 0;
+	}
+
+	if (inst_1->cli.rank < inst_2->cli.rank) {
+		return 1;
+	} else if (inst_1->cli.rank > inst_2->cli.rank) {
```

Comment:
else not needed as you return from if

Reply:
The else covers the case where `inst_1->cli.rank == inst_2->cli.rank`

Reply:
I know, I was thinking rather of:
``
if (a > b) {
  return 1;
}

if (a < b) {
  return -1;
}

return 0;
```

Reply:
Right. 

Are you aware of any branching optimizations or downsides of doing
```c
if (a > b} {
  return 1;
}

if (b < a) {
  return -1;
}

return 0;
```

compared to 
```c
if (a > b} {
  return 1;
} else if (b < a) {
  return -1;
}

return 0;
```

I can change it no problem, even if it does not really change anything. 


With regards to doing `return inst_2->cli.rank - inst_1->cli.rank`, the `qsort` implementation does to provide enough information about whether the value can be less than -1 or greater than 1, but from the Linux `man`page:

> The comparison function must return an integer less than, equal to, or greater  than  zero if  the  first  argument  is considered to be respectively less than, equal to, or greater than the second.  If two members compare as equal, their order in the sorted array is  undefined.

So I guess I can just do the simple subtraction - Will update.